supernode/daemon/mgr/TaskMgr.go

```diff
@@ -0,0 +1,46 @@
+package mgr
+
+import (
+	"context"
+
+	"github.com/dragonflyoss/Dragonfly/apis/types"
+)
+
+// TaskMgr as an interface defines all operations against Task.
+// A Task will store some meta info about the taskFile, pieces and something else.
+// A Task has a one-to-one correspondence with a file on the disk which is identified by taskID.
+type TaskMgr interface {
+	// Register a task represents that someone wants to download a file.
+	// Supernode will get the task file meta and return taskID.
+	// NOTE: If supernode cannot find the task file, the CDN download will be triggered.
+	Register(ctx context.Context, task *types.TaskInfo) (taskID string, err error)
+
+	// Get the task Info with specified taskID.
+	Get(ctx context.Context, taskID string) (*types.TaskInfo, error)
+
+	// List returns the list tasks with filter.
+	List(ctx context.Context, filter map[string]string) ([]*types.TaskInfo, error)
+
+	// CheckTaskStatus check whether the taskID corresponding file exists.
+	CheckTaskStatus(ctx context.Context, taskID string) (bool, error)
```

Comment:
So taskStatus is either `true` or `false`? Also does `TaskInfo` already tell its status some way?

Reply:
This is an interface to quickly check if a file exists, and you can use get for more information, including status.